OneCryptoPkg/OneCryptoBin/OneCryptoBin.c

```diff
@@ -0,0 +1,456 @@
+/** @file
+  This file implements the OneCryptoBin protocol initialization and constructor logic.
+  It sets up the ONE_CRYPTO_PROTOCOL structure with function pointers for cryptographic operations,
+  and provides entry points for MM driver integration.
+
+  Copyright (C) Microsoft Corporation
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include <Uefi.h>
+// #include <Library/StandaloneMmDriverEntryPoint.h> # Intentionally not included
+#include <Library/MmServicesTableLib.h>
+#include <Library/DebugLib.h>
+#include <Library/BaseCryptCrtLib.h>
+#include <Private/OneCryptoDependencySupport.h>
+#include <Library/BaseCryptLib.h>
+#include <Library/TlsLib.h>
+#include <Protocol/OneCrypto.h>
+
+// #if defined(_MSC_VER)
+#define COMMON_EXPORT_API  __declspec(dllexport)
+
+ONE_CRYPTO_MM_CONSTRUCTOR_PROTOCOL  *ProtocolInstance = NULL;
+
+/**
+  Initialize crypto functionality.
+  
+  This function populates the crypto protocol structure with function pointers
+  from BaseCryptLib implementations.
+  
+  @param[in] Crypto  Pointer to crypto protocol structure to initialize.
+**/
+VOID
+EFIAPI
+CryptoInit (
+  IN VOID *Crypto
```

Comment:
This is always called by someone who knows this is OneCrypto, might as well have this param be `ONE_CRYPTO_PROTOCOL`